app/src/androidTest/java/com/github/se/eventradar/qrCode/qrCodeScanTicketUiTest.kt

```diff
@@ -91,6 +90,117 @@ class QrCodeScanTicketUiTest : TestCase(kaspressoBuilder = Kaspresso.Builder.wit
+    }
+  }
+
+  @Test
+  fun screenDisplaysContentElementsCorrectly1() = run {
+    every { mockViewModel.uiState } returns myEventUiState
+    onComposeScreen<QrCodeScanTicketUiScreen>(composeTestRule) {
+      composeTestRule.setContent { QrCodeTicketUi(mockViewModel, mockNavActions) }
+      lazyEventDetails.assertIsDisplayed()
+      eventTitle {
+        assertIsDisplayed()
+        assertTextContains("Test_Event")
+      }
+      eventImage { assertIsDisplayed() }
+      descriptionTitle { assertIsDisplayed() }
+      descriptionContent {
+        assertIsDisplayed()
+        assertTextContains("Test Description")
+      }
+    }
+  }
+
+  @Test
+  fun screenDisplaysContentElementsCorrectly2() = run {
+    every { mockViewModel.uiState } returns myEventUiState
+    onComposeScreen<QrCodeScanTicketUiScreen>(composeTestRule) {
+      composeTestRule.setContent { QrCodeTicketUi(mockViewModel, mockNavActions) }
+      distanceTitle { assertIsDisplayed() }
+      distanceContent { assertIsDisplayed() }
+    }
+  }
+
+  @Test
+  fun screenDisplaysContentElementsCorrectly3() = run {
+    every { mockViewModel.uiState } returns myEventUiState
+    onComposeScreen<QrCodeScanTicketUiScreen>(composeTestRule) {
+      composeTestRule.setContent { QrCodeTicketUi(mockViewModel, mockNavActions) }
+      categoryTitle { assertIsDisplayed() }
+      categoryContent {
+        assertIsDisplayed()
+        assertTextContains("Community")
+      }
+    }
+  }
+
+  @Test
+  fun screenDisplaysContentElementsCorrectly4() =
+      //  Test(timeout = 45.seconds) {
+      run {
+        //      val viewModel = setupViewModelMyEventTab()
+        every { mockViewModel.uiState } returns myEventUiState
+        onComposeScreen<QrCodeScanTicketUiScreen>(composeTestRule) {
+          composeTestRule.setContent { QrCodeTicketUi(mockViewModel, mockNavActions) }
+          dateTitle { assertIsDisplayed() }
+          dateContent { assertIsDisplayed() }
+        }
+      }
+
+  @Test
+  fun screenDisplaysContentElementsCorrectly5() =
+      //  Test(timeout = 45.seconds) {
+      run {
+        //      val viewModel = setupViewModelMyEventTab()
+        every { mockViewModel.uiState } returns myEventUiState
+        onComposeScreen<QrCodeScanTicketUiScreen>(composeTestRule) {
+          composeTestRule.setContent { QrCodeTicketUi(mockViewModel, mockNavActions) }
+          timeTitle { assertIsDisplayed() }
+          timeContent { assertIsDisplayed() }
+        }
+      }
+
+  @Test
+  fun screenDisplaysContentElementsCorrectly6() =
+      //  Test(timeout = 45.seconds) {
+      run {
+        every { mockViewModel.uiState } returns myEventUiState
+        onComposeScreen<QrCodeScanTicketUiScreen>(composeTestRule) {
+          composeTestRule.setContent { QrCodeTicketUi(mockViewModel, mockNavActions) }
+          ticketSoldContent {
+            assertIsDisplayed()
+            assertTextContains("59 tickets sold")
+          }
+        }
+      }
```

Comment:
Question : why did you decided to separate that test into 6 parts ?
I didn't know it existed until late in the project but to separate components being tested in a very long test we can us the `step("message"){...}` function
exemple from our current end to end test
```kotlin
step("Check if all events are present at the start") {
  // Test the UI elements
  eventList { assertIsDisplayed() }
  eventCard { assertIsDisplayed() }
}
step("Filter to only show one event") {
  // Test the UI elements
  searchBarAndFilter { assertIsDisplayed() }
  searchBar { performTextInput("Test 1") }

  eventList { assertIsDisplayed() }
  eventCard { assertIsDisplayed() }
}
```
I think this way you can only have one tests and still be not too hard to debug when one fail

Reply:
sure!